client/test/components/MainBanner/index.test.tsx

```diff
@@ -0,0 +1,25 @@
+import React from 'react';
+import { render, RenderResult } from '@testing-library/react';
+import MainBanner, { IPromotionProps } from '@/components/MainBanner';
+import { PROMOTIONS } from '@mocks/promotions';
+
+describe('MainBanner', () => {
+  const renderMainBanner = (props: IPromotionProps): RenderResult => {
+    return render(<MainBanner promotions={props.promotions} />);
+  };
+
+  describe('with promotion', () => {
+    const promotions = PROMOTIONS;
+
+    test('render main banner img', () => {
+      const { findByAltText } = renderMainBanner({ promotions });
+
+      promotions.forEach(async (promotion) => {
+        const { image } = promotion;
+        const img = await findByAltText(image);
+
+        expect(img).toHaveAttribute('src', image);
+      });
```

Comment:
```suggestion
      for (const promotion of promotions) {
        const { image } = promotion;
        const img = await findByAltText(image);

        expect(img).toHaveAttribute('src', image);
      }
```
테스트가 정상 작동하지 않고 있습니다. 👀 
`forEach` 내부의 `async-await`는 겉에서는 무시되기 때문에 아무것도 실행되지 않아 테스트가 통과되는 것처럼 보일거예요.
`for...of`의 문법을 사용하면 테스트가 제대로 작용해 실패하게 되죠.

`getByAltText`는 img의 alt를 검증하기 때문에, 다른 테스트 방식을 생각해 보시면 좋을 것 같습니다!
가장 쉬운 방법은 alt를 넣는 방법이겠네요 ☺️

Reply:
현우님 코드를 열심히 참고해서 작성했는데도 아주 미흡하군요 :)
머쓱머쓱~~ 피드백 감사합니다!!

Reply:
헉 맞아요! 예전에 저도 api 작성할 때 forEach안의 async-await 때문에 삽질을 한 기억이 있습니다...
지금은 변경된 코드인데 그 때 저는 `Promise.all`로 해결했었어요! 그 때도 현우님이 리뷰해주신거였는데
~갓현우~

Reply:
오오.. forEach 안의 async-await에 대해 한번도 생각해본적 없는데 이번에 알아갑니다 :)